prow/cmd/deck/main.go

```diff
@@ -178,11 +185,11 @@ func traceHandler(h http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		t := time.Now()
 		// Initialize the status to 200 in case WriteHeader is not called
-		trw := &traceResponseWriter{w, http.StatusOK}
+		trw := &traceResponseWriter{ResponseWriter: w, statusCode: http.StatusOK}
 		h.ServeHTTP(trw, r)
 		latency := time.Since(t)
 		deckMetrics.httpRequestDuration.With(
-			prometheus.Labels{"path": r.URL.Path, "method": r.Method, "status": strconv.Itoa(trw.statusCode)}).
+			prometheus.Labels{"path": r.URL.Path, "method": r.Method, "status": strconv.Itoa(trw.statusCode), "size": strconv.Itoa(trw.size)}).
```

Comment:
Should we add `size` at [L158](https://github.com/kubernetes/test-infra/blob/master/prow/cmd/deck/main.go#L158)?

Reply:
I think we want to correlate the size to the request itself? So you could do size by response code or something